contrib/drivers/openstack/cinder/cinder.go

```diff
@@ -412,20 +429,41 @@ func (d *Driver) DeleteSnapshot(req *pb.DeleteVolumeSnapshotOpts) error {
 	return nil
 }
 
+//List VolumeTypes
+func volumeTypeListURL(c *gophercloud.ServiceClient) string {
+	return c.ServiceURL("types")
+}
+
+// StoragePoolPage is a single page of all List results.
+type StoragePoolPage struct {
+	pagination.SinglePageBase
+}
+
+func ListVolumeTypes(client *gophercloud.ServiceClient) pagination.Pager {
+	url := volumeTypeListURL(client)
+	return pagination.NewPager(client, url, func(r pagination.PageResult) pagination.Page {
+		return StoragePoolPage{pagination.SinglePageBase(r)}
+	})
+}
+
 // ListPools
 func (d *Driver) ListPools() ([]*model.StoragePoolSpec, error) {
-	data, err := ioutil.ReadFile(defaultConfPath)
+	volRes, err := ListVolumeTypes(d.blockStoragev2).AllPages()
```

Comment:
Was the pool info never saved in OpenSDS database before?

Reply:
The created pool will be saved to the database, Is there any question with the logic of the code?

Reply:
> Was the pool info never saved in OpenSDS database before?
The created pool will be saved to the database, Is there any question with the logic of the code?

Reply:
Can you submit a cinder.yaml file?  I don't see it in the repo.  If you don't read from the config file any more, how should the config file be changed?

Reply:
Can you show me what are the results of "cinder type-list" and "cinder type-show xxxx" on your setup?  schedulerstats only list pools with pool names, but there are no volume types in that list.  
I don't see how you can find a match this way.